src/messages/messages.test.ts

```diff
@@ -0,0 +1,77 @@
+import { ethers } from "ethers";
+
+import { getConfig, setConfig } from "../config/config";
+import { createBroadcastMessage, hash, sign, verify, DSNPType } from "./messages";
+
+describe("messages", () => {
+  describe("#hash", () => {
+    it("returns the same hash for two identical messages", () => {
+      const message1 = createBroadcastMessage("1", "https://dsnp.org", "0x12345");
+      const message2 = createBroadcastMessage("1", "https://dsnp.org", "0x12345");
+
+      expect(hash(message1)).toEqual(hash(message2));
+    });
+
+    it("returns different hashes for two different messages", () => {
+      const message1 = createBroadcastMessage("1", "https://dsnp.org", "0x12345");
+      const message2 = createBroadcastMessage("2", "https://dsnp.org", "0x12345");
+
+      expect(hash(message1)).not.toEqual(hash(message2));
+    });
+
+    it("returns the same hash for two identical messages with different key orders", () => {
+      const message1 = {
+        contentHash: "0x12345",
+        fromId: "1",
+        type: DSNPType.Broadcast,
+        uri: "https://dsnp.org",
+      };
+      const message2 = {
+        type: DSNPType.Broadcast,
+        uri: "https://dsnp.org",
+        contentHash: "0x12345",
+        fromId: "1",
+      };
+
+      expect(hash(message1)).toEqual(hash(message2));
+    });
+  });
+
+  describe("#sign", () => {
+    beforeAll(async () => {
+      const privateKey = "0x6dcefd57d921dc570e198f6bd9dffc32954ab071184c780770cf4541dd23f68e";
+      await setConfig({
+        ...(await getConfig()),
+        signer: new ethers.Wallet(privateKey),
+      });
+    });
+
+    it("returns a valid signature for a valid private key and message", async () => {
+      const message = createBroadcastMessage("1", "https://dsnp.org", "0x12345");
+
+      expect(await sign(message)).toEqual(
+        "0x9bde6e79c896768bfeed17e6fb7508d4caca83d01a18e5d14a4428e45d8aced33851595e4234e0d4396a8459109e1eb284583f6745a64289723e41aba1b92a8d1c"
```

Comment:
I see that there is effectively a round trip test here, but it'd be nice to actually see the round trip. Confirming that the signature equals this long fixture doesn't provide much value. I'd be nice to see that signing a message, and then verifying with the same signature and a different message produces the wrong public key.

Reply:
These tests use the same public-private key pair and message, so in effect, they already are a round-trip test.

Reply:
Added one anyway: https://github.com/LibertyDSNP/sdk-ts/pull/23/files#diff-a7f6db83f43ab14cc6a3c03d3eeb8f80dd905b6c0e01685ade183f25ee5f5b73R46